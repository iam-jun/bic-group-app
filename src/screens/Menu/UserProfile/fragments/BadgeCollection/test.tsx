import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
