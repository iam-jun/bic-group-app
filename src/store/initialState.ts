import {appInitState} from '~/store/app/reducer';
import {modalInitState} from '~/store/modal/reducer';
import {authInitState} from '~/screens/Auth/redux/reducer';
import {postInitState} from '~/screens/Post/redux/reducer';
import {groupInitState} from '~/screens/Groups/redux/reducer';
import {homeInitState} from '~/screens/Home/redux/reducer';
import {notiInitState} from '~/screens/Notification/redux/reducer';
import {menuInitState} from '~/screens/Menu/redux/reducer';
import {noInternetInitState} from '~/screens/NoInternet/redux/reducer';
import {mentionInputInitState} from '~/beinComponents/inputs/MentionInput/redux/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
};

export default initialState;
