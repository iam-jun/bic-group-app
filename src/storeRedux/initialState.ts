import { appInitState } from '~/storeRedux/app/reducer';
import { modalInitState } from '~/storeRedux/modal/reducer';
import { authInitState } from '~/storeRedux/auth/reducer';
import { postInitState } from '~/storeRedux/post/reducer';
import { groupInitState } from '~/storeRedux/groups/reducer';
import { homeInitState } from '~/storeRedux/home/reducer';
import { notiInitState } from '~/storeRedux/notification/reducer';
import { menuInitState } from '~/storeRedux/menu/reducer';
import { noInternetInitState } from '~/storeRedux/network/reducer';
import { mentionInputInitState } from '~/beinComponents/inputs/MentionInput/redux/reducer';
import { giphyInitState } from './giphy/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
  giphy: giphyInitState,
};

export default initialState;
