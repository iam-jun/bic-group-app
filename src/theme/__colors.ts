// Color system
export const white = '#ffffff';
export const grey1 = '#f4f4f4';
export const grey2 = '#EBEAEC';
export const grey3 = '#dee2e6';
export const grey4 = '#adb5bd';
export const grey5 = '#999999';
export const grey6 = '#777777';
export const grey7 = '#383838';
export const grey8 = '#1e1e1e';
export const grey9 = '#2C2C2C';
export const black = '#000';

export const red = '#ff5148';
export const orange = '#F2711C';
export const yellow = '#FBBD08';
export const olive = '#B5CC18';
export const green = '#4AD498';
export const teal = '#00B5AD';
export const blue = '#2185D0';
export const violet = '#6435C9';
export const purple = '#A333C8';
export const pink = '#E03997';
export const brown = '#A5673F';
export const accent = '#6600cc';
export const primary = '#220b3b';

export default {
  dark: {
    colors: {
      accent: '#6600cc',
      native: '#2A2C36',
      backdrop: 'rgba(0, 0, 0, 0.5)',
      background: grey7,
      disabled: 'rgba(255, 255, 255, 0.38)',
      error: '#E84D4B',
      lightRed: '#E84D4B',
      notification: '#ff80ab',
      onBackground: '#FFFFFF',
      onSurface: '#FFFFFF',
      placeholder: 'rgba(255, 255, 255, 0.54)',
      primary: '#fff',
      surface: black,
      text: '#FFFFFF',
      bgColor: grey8,
      bgColorSecondary: '#220b3b',
      warning: yellow,
      innerButton: '#FFFFFF',
      buttonBackground: '#220b3b',
      hightLight: '#59bbc4',
      textSecondary: '#7fe5f0',
      tag: grey7,
      red,
      orange,
      yellow,
      olive,
      green,
      blue,
      teal,
      violet,
      purple,
      pink,
      brown,
      grey1,
      grey2,
      grey3,
      grey4,
      grey5,
      grey6,
      grey7,
      grey8,
      grey9,
      black,
      //TabBar
      tabBarBackground: black,
      activeColor: '#F7F7F7',
      inactiveColor: grey6,
      // ViewLabel
      ViewLabel: {
        color: grey6,
        colorHeading: white,
      },
      secondary: grey6,
      white,
      greyOutline: '#bbb',
      searchBg: '#303337',
      listItemBg: grey8,
      success: green,
      divider: grey7,
      border: grey9,
      icon: grey7,
      avatar: '#b28a5c',
      platform: {
        ios: {
          primary: '#007aff',
          secondary: '#5856d6',
          success: '#4cd964',
          error: '#ff3b30',
          warning: '#ffcc00',
        },
        android: {
          primary: '#2196f3',
          secondary: '#9C27B0',
          success: '#4caf50',
          error: '#f44336',
          warning: '#ffeb3b',
        },
      },
      headerStyle: {
        backgroundColor: '#1e1f28',
        headerTintColor: '#ffffff',
      },
      facebook: '#2D88FF',
      google: '#FF2828',
      apple: '#FFFFFF',
    },
  },
  light: {
    colors: {
      accent: '#6600cc',
      native: '#E5E5E5',
      backdrop: 'rgba(0, 0, 0, 0.5)',
      background: '#f6f6f6',
      disabled: 'rgba(0, 0, 0, 0.26)',
      error: '#E84D4B',
      lightRed: '#E84D4B',
      notification: '#f50057',
      onBackground: '#252D42',
      onSurface: '#000000',
      placeholder: 'rgba(0, 0, 0, 0.54)',
      primary: '#220b3b',
      surface: '#ffffff',
      text: '#252D42',
      bgColor: white,
      bgColorSecondary: '#f6f6f6',
      warning: yellow,
      icon: grey3,
      avatar: '#707f9f',
      innerButton: '#368C8B',
      buttonBackground: '#EBEAEC',
      hightLight: '#59bbc4',
      tag: grey3,
      textSecondary: '#00b9cf',
      red,
      orange,
      yellow,
      olive,
      green,
      blue,
      teal,
      violet,
      purple,
      pink,
      brown,
      grey9,
      //TabBar
      tabBarBackground: '#fff',
      activeColor: '#220b3b',
      inactiveColor: '#adb5bd',
      // ViewLabel
      ViewLabel: {
        color: grey5,
        colorHeading: black,
      },
      secondary: grey6,
      white: white,
      grey1: grey1,
      grey2: grey2,
      grey3: grey3,
      grey4: grey4,
      grey5: grey5,
      grey6: grey6,
      grey7: grey7,
      grey8: grey8,
      black: black,

      greyOutline: '#bbb',
      searchBg: '#303337',
      listItemBg: white,
      success: green,
      // Darker color if hairlineWidth is not thin enough
      divider: grey2,
      border: grey2,
      platform: {
        ios: {
          primary: '#007aff',
          secondary: '#5856d6',
          success: '#4cd964',
          error: '#ff3b30',
          warning: '#ffcc00',
        },
        android: {
          primary: '#2196f3',
          secondary: '#9C27B0',
          success: '#4caf50',
          error: '#f44336',
          warning: '#ffeb3b',
        },
      },
      headerStyle: {
        backgroundColor: '#ffffff',
        headerTintColor: '#000000',
      },
      facebook: '#2D88FF',
      google: '#FF2828',
      apple: '#000000',
    },
  },
};
