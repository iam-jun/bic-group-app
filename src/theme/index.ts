export { default as fonts } from './fonts';
export { default as spacing } from './spacing';
export { default as dimension } from './dimension';
export { default as shadow } from './shadow';
