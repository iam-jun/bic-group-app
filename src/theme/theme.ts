import { DefaultTheme, DarkTheme, ExtendedTheme } from '@react-navigation/native';
import elevations from './elevations';

export const light: ExtendedTheme = {
  dark: false,
  colors: {
    ...DefaultTheme.colors,
    black: '#000000',
    white: '#FFFFFF',
    danger: '#F64E60',
    success: '#13B055',
    warning: '#FFBC1F',

    neutral: '#FFFFFF',
    neutral1: '#F8F9FF',
    neutral2: '#F1F2F8',
    neutral5: '#E3E6F2',
    neutral10: '#D6D9EB',
    neutral20: '#BAC0DE',
    neutral30: '#9FA6D1',
    neutral40: '#353E6E',
    neutral50: '#2E3660',
    neutral60: '#282E53',
    neutral70: '#212745',
    neutral80: '#181C32',
    neutral90: '#0D0F1C',

    gray1: '#F2F5F8',
    gray5: '#EAEDF2',
    gray10: '#DDDFE4',
    gray20: '#D1D4DB',
    gray30: '#BABEC9',
    gray40: '#A4A9B7',
    gray50: '#767D93',
    gray60: '#515667',
    gray70: '#3F4350',
    gray80: '#2D3039',
    gray90: '#1B1D22',

    purple1: '#FBF8FF',
    purple2: '#F4EFFB',
    purple5: '#E9DFF6',
    purple10: '#C9AFE9',
    purple20: '#B48FE0',
    purple30: '#9E6FD8',
    purple40: '#884FCF',
    purple50: '#7335C0',
    purple60: '#602CA0',

    blue1: '#F5FAFF',
    blue2: '#EBF4FF',
    blue5: '#D6EAFF',
    blue10: '#ADD5FF',
    blue20: '#85C0FF',
    blue30: '#5CABFF',
    blue40: '#3396FF',
    blue50: '#007AFF',
    blue60: '#006CE0',

    red1: '#FEF3F4',
    red2: '#FEECED',
    red5: '#FDD8DC',
    red10: '#FCC5CA',
    red20: '#F1999F',
    red30: '#F0878D',
    red40: '#F64E60',
    red50: '#F42A3E',
    red60: '#E90C22',

    green1: '#FAFFFC',
    green2: '#EFFDF4',
    green5: '#E5F9EC',
    green10: '#D0F1DB',
    green20: '#8ED6A4',
    green30: '#74C18C',
    green40: '#56B374',
    green50: '#13B055',
    green60: '#0E9B49',

    slate1: '#EEEDFD',
    slate5: '#DDDAFB',
    slate10: '#CCC8F9',
    slate20: '#AAA3F5',
    slate30: '#8A81F7',
    slate40: '#7B71F0',
    slate50: '#6559EE',
    slate60: '#4334EA',

    violet1: '#F1F1F9',
    violet5: '#E3E4F3',
    violet10: '#D4D6EC',
    violet20: '#B8BBE0',
    violet30: '#A1A6DC',
    violet40: '#959AD1',
    violet50: '#7F85C7',
    violet60: '#636ABB',

    turquoise1: '#EDFDFC',
    turquoise5: '#DBFAF9',
    turquoise10: '#B7F5F2',
    turquoise20: '#93F0EC',
    turquoise30: '#5DE9E2',
    turquoise40: '#28E2D8',
    turquoise50: '#1BC5BD',
    turquoise60: '#18B4AC',

    transparent: 'rgba(255,255,255,0)',
    transparent1: 'rgba(17, 21, 26, 0.6)',
  },
  elevations: elevations.light,
};

export const dark: ExtendedTheme = {
  dark: true,
  colors: {
    ...DarkTheme.colors,
    black: '#000000',
    white: '#FFFFFF',
    danger: '#F64E60',
    success: '#13B055',
    warning: '#FFBC1F',

    neutral: '#FFFFFF',
    neutral1: '#F8F9FF',
    neutral2: '#F1F2F8',
    neutral5: '#E3E6F2',
    neutral10: '#D6D9EB',
    neutral20: '#BAC0DE',
    neutral30: '#9FA6D1',
    neutral40: '#353E6E',
    neutral50: '#2E3660',
    neutral60: '#282E53',
    neutral70: '#212745',
    neutral80: '#181C32',
    neutral90: '#0D0F1C',

    gray1: '#F2F5F8',
    gray5: '#EAEDF2',
    gray10: '#DDDFE4',
    gray20: '#D1D4DB',
    gray30: '#BABEC9',
    gray40: '#A4A9B7',
    gray50: '#767D93',
    gray60: '#515667',
    gray70: '#3F4350',
    gray80: '#2D3039',
    gray90: '#1B1D22',

    purple1: '#FBF8FF',
    purple2: '#F4EFFB',
    purple5: '#E9DFF6',
    purple10: '#C9AFE9',
    purple20: '#B48FE0',
    purple30: '#9E6FD8',
    purple40: '#884FCF',
    purple50: '#7335C0',
    purple60: '#602CA0',

    blue1: '#F5FAFF',
    blue2: '#EBF4FF',
    blue5: '#D6EAFF',
    blue10: '#ADD5FF',
    blue20: '#85C0FF',
    blue30: '#5CABFF',
    blue40: '#3396FF',
    blue50: '#007AFF',
    blue60: '#006CE0',

    red1: '#FEF3F4',
    red2: '#FEECED',
    red5: '#FDD8DC',
    red10: '#FCC5CA',
    red20: '#F1999F',
    red30: '#F0878D',
    red40: '#F64E60',
    red50: '#F42A3E',
    red60: '#E90C22',

    green1: '#FAFFFC',
    green2: '#EFFDF4',
    green5: '#E5F9EC',
    green10: '#D0F1DB',
    green20: '#8ED6A4',
    green30: '#74C18C',
    green40: '#56B374',
    green50: '#13B055',
    green60: '#0E9B49',

    slate1: '#EEEDFD',
    slate5: '#DDDAFB',
    slate10: '#CCC8F9',
    slate20: '#AAA3F5',
    slate30: '#8A81F7',
    slate40: '#7B71F0',
    slate50: '#6559EE',
    slate60: '#4334EA',

    violet1: '#F1F1F9',
    violet5: '#E3E4F3',
    violet10: '#D4D6EC',
    violet20: '#B8BBE0',
    violet30: '#A1A6DC',
    violet40: '#959AD1',
    violet50: '#7F85C7',
    violet60: '#636ABB',

    turquoise1: '#EDFDFC',
    turquoise5: '#DBFAF9',
    turquoise10: '#B7F5F2',
    turquoise20: '#93F0EC',
    turquoise30: '#5DE9E2',
    turquoise40: '#28E2D8',
    turquoise50: '#1BC5BD',
    turquoise60: '#18B4AC',

    transparent: 'rgba(255,255,255,0)',
    transparent1: 'rgba(17, 21, 26, 0.6)',
  },
  elevations: elevations.dark,
};

export default {
  light,
  dark,
};
