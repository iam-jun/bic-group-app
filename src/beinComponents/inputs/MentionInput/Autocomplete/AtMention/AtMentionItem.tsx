import React from 'react';
import {StyleSheet, TouchableOpacity, View} from 'react-native';
import {useTheme} from 'react-native-paper';

import Avatar from '~/beinComponents/Avatar';
import Text from '~/beinComponents/Text';
import images from '~/resources/images';
import {ITheme} from '~/theme/interfaces';
import spacing from '~/theme/spacing';

interface Props {
  testID?: string;
  item: any;
  onPress: (item: any) => void;
}

const AtMentionItem = ({item, onPress}: Props) => {
  const theme = useTheme() as ITheme;
  const styles = createStyles(theme);

  const _onPressItem = () => {
    onPress(item);
  };

  const renderMentionAll = () => {
    return (
      <View>
        <TouchableOpacity
          testID="at_mention_item.item_all"
          onPress={_onPressItem}>
          <View style={styles.mentionAll}>
            <Text.ButtonBase style={styles.textMentionAll}>
              @all
            </Text.ButtonBase>
            <Text.Subtitle useI18n>common:title_mention_all</Text.Subtitle>
          </View>
        </TouchableOpacity>
      </View>
    );
  };

  if (item.username === 'all') return renderMentionAll();

  return (
    <View testID={'at_mention_item'} style={styles.mentionContainer}>
      <TouchableOpacity
        style={styles.container}
        onPress={_onPressItem}
        testID={'at_mention_item.touchable'}>
        <Avatar.Medium
          style={styles.avatar}
          source={item.avatar || item.icon}
          placeholderSource={images.img_user_avatar_default}
        />
        <Text>{item.name || item.fullname}</Text>
      </TouchableOpacity>
    </View>
  );
};

const createStyles = (theme: ITheme) => {
  const {colors} = theme;
  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    avatar: {
      marginHorizontal: spacing.margin.base,
      marginVertical: spacing.margin.small,
    },
    mentionContainer: {
      backgroundColor: colors.background,
    },
    mentionAll: {
      flexDirection: 'row',
      padding: spacing?.padding.base,
      alignItems: 'center',
    },
    textMentionAll: {
      marginEnd: spacing?.margin.base,
    },
  });
};

export default AtMentionItem;
