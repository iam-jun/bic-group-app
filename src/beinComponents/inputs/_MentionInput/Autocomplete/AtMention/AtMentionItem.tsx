import React from 'react';
import {StyleSheet, TouchableOpacity} from 'react-native';
import {useTheme} from 'react-native-paper';
import {useDispatch} from 'react-redux';
import Avatar from '~/beinComponents/Avatar';
import Div from '~/beinComponents/Div';
import Text from '~/beinComponents/Text';
import {AT_MENTION_REGEX} from '~/constants/autocomplete';
import {useKeySelector} from '~/hooks/selector';
import images from '~/resources/images';
import {ITheme} from '~/theme/interfaces';
import actions from '../../redux/actions';

const AtMentionItem = ({item}: any) => {
  const dispatch = useDispatch();
  const {text, cursorPosition, highlightItem} = useKeySelector('mentionInput');

  const theme = useTheme() as ITheme;
  const {colors} = theme;
  const styles = createStyles(theme);

  const backgroundColor =
    highlightItem?.id && item?.id === highlightItem?.id
      ? colors.placeholder
      : colors.background;

  const onHoverItem = () => {
    dispatch(actions.sethHighlightItem(item));
  };

  const onLeaveItem = () => {
    dispatch(actions.sethHighlightItem(null));
  };

  const completeMention = (mention: string) => {
    //    const {cursorPosition, isSearch, onChangeText, value} = this.props;
    const mentionPart = text.substring(0, cursorPosition);

    let completedDraft = mentionPart.replace(AT_MENTION_REGEX, `@${mention} `);

    if (text.length > cursorPosition) {
      completedDraft += text.substring(cursorPosition);
    }

    dispatch(actions.setText(completedDraft));
    dispatch(actions.setData([]));
  };

  const _onPressItem = () => {
    completeMention(item.username);
  };

  return (
    <Div
      style={{backgroundColor}}
      onMouseOver={onHoverItem}
      onMouseLeave={onLeaveItem}>
      <TouchableOpacity style={styles.container} onPress={_onPressItem}>
        <Avatar.Medium
          style={styles.avatar}
          source={item.avatar || item.icon}
          placeholderSource={images.img_user_avatar_default}
        />
        <Text>{item.name || item.fullname}</Text>
      </TouchableOpacity>
    </Div>
  );
};

const createStyles = (theme: ITheme) => {
  const {spacing} = theme;
  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    avatar: {
      marginHorizontal: spacing.margin.base,
      marginVertical: spacing.margin.small,
    },
  });
};

export default AtMentionItem;
