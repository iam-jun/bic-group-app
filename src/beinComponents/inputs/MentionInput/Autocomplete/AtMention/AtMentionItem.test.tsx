import React from 'react';
import {StyleSheet} from 'react-native';
import {cleanup} from '@testing-library/react-native';

import {configureStore, renderWithRedux, fireEvent} from '~/test/testUtils';
import initialState from '~/store/initialState';
import AtMentionItem from './AtMentionItem';
import {colors} from '~/theme';

afterEach(cleanup);

describe('AtMentionItem component', () => {
  const baseProps = {
    showSpectialItems: false,
    item: {
      username: 'test',
      name: 'test',
    },
    onPress: jest.fn(),
  };

  const mockStore = configureStore([]);

  it(`renders correctly`, async () => {
    const store = mockStore(initialState);
    const wrapper = renderWithRedux(<AtMentionItem {...baseProps} />, store);

    const rendered = wrapper.toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it(`should show "AtMentionItem" with highlight item`, async () => {
    const storeData = {...initialState};
    const highlightItem: any = {
      username: 'test',
      name: 'test',
    };
    storeData.mentionInput.data = [
      {
        username: 'abc',
        name: 'abc',
      },
      highlightItem,
    ] as any;
    storeData.mentionInput.highlightItem = highlightItem;
    const store = mockStore(storeData);
    const props = {
      ...baseProps,
      item: highlightItem,
    };
    const wrapper = renderWithRedux(<AtMentionItem {...props} />, store);
    const component = wrapper.getByTestId('at_mention_item');
    expect(component).not.toBeNull();
    const flattenedStyle = StyleSheet.flatten(component.props.style);
    expect(flattenedStyle.backgroundColor).toBe(colors.light.colors.neutral5);
  });

  it(`should show "AtMentionItem" with item all`, async () => {
    const store = mockStore(initialState);
    const props = {
      ...baseProps,
      showSpectialItems: true,
    };
    const wrapper = renderWithRedux(<AtMentionItem {...props} />, store);
    const component = wrapper.getByTestId('at_mention_item');

    expect(component).not.toBeNull();
  });

  it(`should call "_onPressItem"`, async () => {
    const onPress = jest.fn();
    const storeData = {...initialState};
    storeData.mentionInput.data = [
      {
        username: 'test',
        name: 'test',
      },
    ] as any;
    const store = mockStore(storeData);
    const props = {
      ...baseProps,
      onPress,
    };
    const wrapper = renderWithRedux(<AtMentionItem {...props} />, store);
    const component = wrapper.getByTestId('at_mention_item.touchable');
    expect(component).not.toBeNull();
    fireEvent.press(component);
    expect(onPress).toBeCalled();
  });
});
