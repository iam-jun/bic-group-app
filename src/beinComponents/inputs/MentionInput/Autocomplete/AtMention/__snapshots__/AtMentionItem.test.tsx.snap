// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtMentionItem component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
  testID="at_mention_item"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
      }
    }
    testID="at_mention_item.touchable"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 6,
            "height": 40,
            "width": 40,
          },
          Object {
            "marginHorizontal": 12,
            "marginVertical": 8,
          },
        ]
      }
      testID="avatar"
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 6,
              "height": 40,
              "width": 40,
            },
            Object {
              "backgroundColor": "#A4A9B7",
            },
          ]
        }
        testID="avatar_container"
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "borderRadius": 6,
                  "height": 40,
                  "width": 40,
                },
                Object {
                  "borderColor": "#F2F5F8",
                  "borderWidth": 2,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
              ],
            ]
          }
        >
          <FastImageView
            onFastImageError={[Function]}
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../assets/images/user_avatar_default.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="avatar.image"
          />
        </View>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Light",
            "fontSize": 14,
          },
          Object {},
          undefined,
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
